src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -100,24 +115,30 @@ def run_relevance_eval(
     ].map(list)
 
     exploded_df = filtered_df.explode(retrieved_documents_column_name, ignore_index=False)
-    exploded_df[retrieved_document_text_column_name] = [
-        document_data["document.content"] if document_data is not None else None
-        for document_data in exploded_df[retrieved_documents_column_name]
-    ]
+    if trace_data:
```

Comment:
people will not know to set this flag - we should detect it or rely on composition so that the function is more pure.

Reply:
This code-path needs a test to ensure the df in is the same as the df out.

Reply:
I like the idea of detecting, good idea.